File: tutorials/java/Welcome03.java
/*
 * Arrays of objects
 */

import core.data.*;
import java.util.Scanner;

public class Welcome03 {
   public static void main(String[] args) {
      DataSource ds = DataSource.connect("http://weather.gov/xml/current_obs/index.xml").load();
      WeatherStation[] allstns = ds.fetchArray("WeatherStation", "station/station_name", 
                                               "station/station_id", "station/state",
                                               "station/latitude", "station/longitude");
      System.out.println("Total stations: " + allstns.length);
      
      Scanner sc = new Scanner(System.in);
      System.out.println("Enter a state abbreviation: ");
      String state = sc.next();
      System.out.println("Stations in " + state);
      for (WeatherStation ws : allstns) {
         if (ws.isLocatedInState(state)) {
            System.out.println("  " + ws.getId() + ": " + ws.getName());
         }
      }
   }
}

File: tutorials/python/welcome03.py

from sinbad import *

def main():
    ds = Data_Source.connect_load("http://weather.gov/xml/current_obs/index.xml")

    all_stns = ds.fetch("station_name", "station_id", "state",
                        "latitude", "longitude",
                        base_path = "station")
    print("Total stations:", len(all_stns))

    state_of_interest = input("Enter a state abbreviation: ")

    print("Stations in", state_of_interest)
    for ws in all_stns:
        if ws["state"] == state_of_interest:
            print("   " + ws["station_id"] + ": " + ws["station_name"])


main()
